### ISSUE TITLE
Support compressed texture uploads to GPU

### ISSUE THREAD

**Issue by guilt** [liballeg/allegro5#1033]
Essentially, this is the matrix of texture compression. This means that we may not be able to support all operations on these compressed textures/bitmaps, but they'd be much faster to upload to the GPU, and give us plenty headroom. It is quite helpful on mobile devices and such.

* iOS: PVRTC
* tvOS: ASTC
* Android: ETC1, ETC2, ATC
* Mac/PC/Linux: DXT1, DXT5, BC7, BC6H (HDR)

**Comment by guilt** [comment]
Also see #559

**Comment by beoran** [comment]
We already have some support for DDS files with textures compressed in the DXT1, DXT3 and DXT5 formats. If the other formats are unencumbered by patent, then we could also implement those. See e.g.
https://github.com/liballeg/allegro5/blob/5.2.4/addons/image/dds.c for more information.

**Comment by guilt** [comment]
Actually, we error out on those 3 in the file you just pointed out :)

**Comment by SiegeLord** [comment]
Yeah, we already have some support for (blocked) compressed textures. See `al_lock_bitmap_blocked` for the method to convert to GPU. For OpenGL this uses `glCompressedTexSubImage2D` under the hood. We'll have to think how to support non-blocked compressed textures and things like `ASTC` which have variable block size.

**Comment by guilt** [comment]
ASTC offers varying block sizes, but the block size is still fixed within a file; Just that the total number of bytes per block is a constant = 14 bytes. So, the bits per pixel are different.

Source, code in : https://arm-software.github.io/opengl-es-sdk-for-android/astc_textures.html

The usecase for something like this is to just allow the entire bitmap to be uploaded as-is and compressed into the GPU, and not assumed to be efficient for user modification; This may be like using `glCompressedTexImage2D` for uploading the texture.
